engine/executor/dm/worker.go

```diff
@@ -0,0 +1,242 @@
+	dmconfig "github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/metadata"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/runtime"
+	"github.com/pingcap/tiflow/engine/lib"
+	libModel "github.com/pingcap/tiflow/engine/lib/model"
+	"github.com/pingcap/tiflow/engine/lib/registry"
+	"github.com/pingcap/tiflow/engine/model"
+	dcontext "github.com/pingcap/tiflow/engine/pkg/context"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+	"github.com/pingcap/tiflow/engine/pkg/externalresource/broker"
+	"github.com/pingcap/tiflow/engine/pkg/p2p"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// RegisterWorker is used to register dm task to global registry
+func RegisterWorker() {
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMDump, newWorkerFactory(lib.WorkerDMDump))
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMLoad, newWorkerFactory(lib.WorkerDMLoad))
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMSync, newWorkerFactory(lib.WorkerDMSync))
+}
+
+// workerFactory create dm task
+type workerFactory struct {
+	workerType libModel.WorkerType
+}
+
+// newWorkerFactory creates abstractFactory
+func newWorkerFactory(workerType libModel.WorkerType) *workerFactory {
+	return &workerFactory{workerType: workerType}
+}
+
+// DeserializeConfig implements WorkerFactory.DeserializeConfig
+func (f workerFactory) DeserializeConfig(configBytes []byte) (registry.WorkerConfig, error) {
+	cfg := &dmconfig.SubTaskConfig{}
+	err := cfg.Decode(string(configBytes), true)
+	return cfg, err
+}
+
+// NewWorkerImpl implements WorkerFactory.NewWorkerImpl
+func (f workerFactory) NewWorkerImpl(ctx *dcontext.Context, workerID libModel.WorkerID, masterID libModel.MasterID, conf lib.WorkerConfig) (lib.WorkerImpl, error) {
+	return newDMWorker(ctx, masterID, f.workerType, conf.(*dmconfig.SubTaskConfig)), nil
+}
+
+// dmWorker implements methods for lib.WorkerImpl
+type dmWorker struct {
+	lib.BaseWorker
+
+	unitHolder   unitHolder
+	messageAgent dmpkg.MessageAgent
+
+	mu                 sync.RWMutex
+	cfg                *dmconfig.SubTaskConfig
+	storageWriteHandle broker.Handle
+	stage              metadata.TaskStage
+	workerType         libModel.WorkerType
+	taskID             string
+	masterID           libModel.MasterID
+}
+
+func newDMWorker(ctx *dcontext.Context, masterID libModel.MasterID, workerType lib.WorkerType, cfg *dmconfig.SubTaskConfig) *dmWorker {
+	w := &dmWorker{
+		cfg:        cfg,
+		stage:      metadata.StageInit,
+		workerType: workerType,
+		taskID:     cfg.SourceID,
+		masterID:   masterID,
+		unitHolder: newUnitHolderImpl(workerType, cfg),
+	}
+
+	// nolint:errcheck
+	ctx.Deps().Construct(func(m p2p.MessageHandlerManager) (p2p.MessageHandlerManager, error) {
+		w.messageAgent = dmpkg.NewMessageAgentImpl(w.taskID, w, m)
```

Comment:
For all functions called by commandHandler, please add a comment like "FuncXXX is called by refection of commandHandler", otherwise reader will not find how are they called.